include/core/Partition_Parser.hpp

```diff
@@ -93,29 +93,35 @@ class Partitions_Parser {
                 }
                 part_data.nexus_ids = nexus_ids;
                 nexus_ids.clear();
-                if( part.at("remote-connections").get_type() == geojson::PropertyType::List ) {
+
+                if( part.at("remote-connections").get_type() == geojson::PropertyType::List ) 
+                {
                     //It is valid to have no remote connections, but the backend property tree parser
                     //can't represent empty lists/objects, so it turns into an ampty string (which is iterable)
                     //so we check to ensure the remote connections are a list type (not string) before we attempt
                     //to process the remote-connections.  If they are empty, this step gets skipped entirely.
                     //Get remote-connections and set the corresponding part_data struct member
+                    
                     for (auto &remote_conn : part.at("remote-connections").as_list())
                     {
-                        remote_mpi_rank = remote_conn.at("mpi-rank").as_natural_number();
-                        remote_nex_id = remote_conn.at("nex-id").as_string();
-                        remote_cat_id = remote_conn.at("cat-id").as_string();
-                        direction = remote_conn.at("cat-direction").as_string();
-                        tmp_tuple = std::make_tuple(remote_mpi_rank, remote_nex_id, remote_cat_id, direction);
-                        remote_conn_vec.push_back(tmp_tuple);
+                        if ( remote_conn.get_type() != geojson::PropertyType::String )
+                        {
+                            remote_mpi_rank = remote_conn.at("mpi-rank").as_natural_number();
+                            remote_nex_id = remote_conn.at("nex-id").as_string();
+                            remote_cat_id = remote_conn.at("cat-id").as_string();
+                            direction = remote_conn.at("cat-direction").as_string();
+                            tmp_tuple = std::make_tuple(remote_mpi_rank, remote_nex_id, remote_cat_id, direction);
+                            remote_conn_vec.push_back(tmp_tuple);
+                        }
                     }
-                }
-                part_data.remote_connections = remote_conn_vec;
-                remote_conn_vec.clear();
+                    part_data.remote_connections = remote_conn_vec;
+                    remote_conn_vec.clear();
 
-                //Push part_data struct the vector
-                partition_ranks.push_back(part_data);
+                    //Push part_data struct the vector
+                    partition_ranks.push_back(part_data);
```

Comment:
Now this and the increment below only happens if the partition has a `remote-connections` key--which however may be empty. Was it previously valid to omit the `remote-connections` key entirely? (Just checking)

Reply:
The key has to be present, at least how current parsing works, but it can be empty. We should probably make it fully optional.